src/json_parser/json_handler.py

```diff
@@ -0,0 +1,112 @@
+import json
+from jsonschema import validate
+
+from src.json_parser.json_schemas import SHAPES_SCHEMA, COMPOSITE_SHAPES_SCHEMA, ACTIONS_SCHEMA
+
+
+class JsonHandler:
+    """
+    Class to handle JSON file operations, including reading, parsing, and validating JSON data.
+    """
+
+    def __init__(self, json_path):
+        """
+        Initialize the JsonHandler with the path to the JSON file.
+
+        :param json_path: Path to the JSON file.
+        """
+        self.json_path = json_path
+        self.json_data = self.read_json()
+
+    def read_json(self) -> dict:
+        """
+        Read the JSON file and return its content as a dictionary.
+
+        :return: Dictionary containing the JSON data.
+        """
+        with open(self.json_path, 'r') as json_file:
+            return json.load(json_file)
+
+    def get_shapes(self) -> list:
+        """
+        Retrieve the list of shapes from the JSON data.
+
+        :return: List of shapes.
+        """
+        try:
+            return self.json_data['shapes']
+        except KeyError:
+            return []
+
+    def get_composite_shapes(self) -> list:
+        """
+        Retrieve the list of composite shapes from the JSON data.
+
+        :return: List of composite shapes.
+        """
+        try:
+            return self.json_data['composite_shapes']
+        except KeyError:
+            return []
+
+    def get_actions(self) -> list:
+        """
+        Retrieve the list of actions from the JSON data.
+
+        :return: List of actions.
+        """
+        try:
+            return self.json_data['actions']
+        except KeyError:
+            return []
+
+    def get_shape_by_id(self, shape_id) -> dict:
+        """
+        Retrieve a shape by its ID from the JSON data.
+
+        :param shape_id: ID of the shape to retrieve.
+        :return: Dictionary representing the shape.
+        """
+        for shape in self.get_shapes():
+            if shape['id'] == shape_id:
```

Comment:
I'm not sure that I understand you.
If the user asks to find specific element in array it can't be O(1).
Note that I loop only the array and not the full file